### PR TITLE
Add tests on tries + fix and accelerate tfdf calculations

### PR DESCRIPTION
**Pull request by warenlg**
In this PR I did 2 things : 
1. I add test_tfdf.py which contains all tests on tries. It uses 3 history examples stored in shcomplete/tests/data
2. I found a mistake in tfdf.py when computing tfdf values. It speeds the computation from 1h to 1m

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ _install: &_install
   - pip install -r requirements.txt codecov
   - pip install -e .
 _coverage: &_coverage
-  - SCRIPT="coverage run -m unittest discover"
+  - SCRIPT="python3 -m unittest discover shcomplete/"
```
Comment (vmarkovtsev) [inline .travis.yml]: Have you tried `coverage run -m unittest discover shcomplete` ? You do not collect any coverage information in your current proposal.

File: shcomplete/repos.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# coding=utf-8
```
Comment (vmarkovtsev) [inline shcomplete/repos.py]: That's not needed for Python3

File: shcomplete/tfdf.py [hunk 3]
```diff
@@ -1,11 +1,9 @@
+from __future__ import division
```
Comment (vmarkovtsev) [inline shcomplete/tfdf.py]: That's not needed for Python3

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

**Comment by vmarkovtsev** [comment]
The rest LGTM

File: shcomplete/tfdf.py (hunk 3 above)
Comment (warenlg) [reply shcomplete/tfdf.py]: Ok I understand now, when I ran converage on my computer, It did with python2, that's why I had those mistake. But on github Travis understands that it should "run coverage" with python3

**Pull request closed without merging**